pyfar/classes/coordinates.py

```diff
@@ -1907,11 +2452,54 @@ def cyl2cart(azimuth, height, radius):
     To ensure proper handling of the azimuth angle, the ``arctan2``
     implementation from numpy is used.
     """
+    azimuth = np.atleast_1d(azimuth)
+    height = np.atleast_1d(height)
+    radius = np.atleast_1d(radius)
+
     x = radius * np.cos(azimuth)
     y = radius * np.sin(azimuth)
     if isinstance(height, np.ndarray):
         z = height.copy()
     else:
         z = height
 
+    x[np.abs(x) < np.finfo(x.dtype).eps] = 0
+    y[np.abs(y) < np.finfo(y.dtype).eps] = 0
+    z[np.abs(z) < np.finfo(x.dtype).eps] = 0
+
     return x, y, z
+
+
+def _check_range_angle(angles, lower_limit, upper_limit):
+    """colatitude will be clipped to its range if deviations are below 2 eps
+    for 32 bit float numbers otherwise Error is raised.
+
+    Parameters
+    ----------
+    angles : np.ndarray
+        input array angle
+    lower_limit : float
+        lower limit for angle definition
+    upper_limit : float
+        upper limit for angle definition
+
+    Returns
+    -------
+    angles : np.ndarray
+        clipped input angles
+    """
+    if any(angles < lower_limit):
+        mask = angles < lower_limit
+        eps = np.finfo(float).eps
+        if any(angles[mask]+2*eps < lower_limit):
+            raise ValueError(
+                f'one or more angles are below {lower_limit} including 2 eps')
+        angles[mask] = lower_limit
+    if any(angles > upper_limit):
+        mask = angles > upper_limit
+        eps = np.finfo(float).eps
+        if any(angles[mask] + 2*eps > upper_limit):
+            raise ValueError(
+                f'one or more angles are above {upper_limit} including 2 eps')
+        angles[mask] = upper_limit
+    return angles
```

Comment:
This function is not limited to angles, but could be used for any kind of number in a given interval, right? It could be named in a more general manner.